backend/piece.py

```diff
@@ -0,0 +1,91 @@
+from backend.position import Position
+
+
+class Piece(Position):
+    def __init__(self, x=9, y=9, color="", is_king=False):
+        Position.__init__(self, x, y)
+        self.color = color
+        self.is_king = is_king
+
+    def get_color(self):
+        return self.color
+
+    def set_color(self, color):
+        self.color = color
+
+    def get_is_king(self):
+        return self.is_king
+
+    def set_is_king(self):
+        if (self.get_x() == 0 and self.get_color() == "black") or (self.get_x() == 7 and self.get_color() == "red"):
+            self.is_king = True
+
+    def is_player_piece(self):
+        if (self.color == "red") or (self.color == "black"):
+            return True
+        return False
+
+    def get_all_piece_information(self):
+        return {"position": self.get_position(),
+                "color": self.get_color(),
+                "is_king": self.get_is_king()}
+
+    def possible_move(self):
```

Comment:
improve the whole move system, try another way that doens't cause a code to repeat itself, or by adding a single variable to short the code by half